.circleci/config.yml

```diff
@@ -54,9 +54,8 @@ jobs:
           command: |
               . venv/bin/activate
               python --version
-              python -m unittest tests.test_dash_html_components
-              python -m unittest tests.test_integration
-              python -m unittest tests.test_dash_import
+              npm run test
+              npm run lint
```

Comment:
I'd do the opposite and have linting run as early as possible (before `npm run clean`), to prevent executing a long-ish build and tests for nothing.

Reply:
Or in this case, I guess the .js files are also generated -- so after the generation.

Reply:
> to prevent executing a long-ish build and tests for nothing.

I could move it, but I wouldn't want linter errors to prevent running the rest of the tests. The tests aren't that expensive, seems to me it's worth knowing about all failures together. Personally linter errors are about the last thing I deal with - in the unusual case that they're not caught automatically by my editor, forcing me to fix them right away.

Reply:
Fair enough for the ordering. I had scenarios like the table where running the tests takes up to 15 minutes :)

Reply:
Actually, I don't think you'll get all the errors as-is. If tests fail, the job will terminate before linting happens. You would need a job for tests and one for linting to always get both sources of failure.

Reply:
Nice! That should also work.

Reply:
OK, my bash-fu is not the best, but I couldn't find a straight shell solution that worked the way I wanted... everything I tried either stopped on first failure or ignored errors if something later passed. `npm-run-all` and its `run-s` command ([with the `-c` option](https://github.com/mysticatea/npm-run-all/blob/HEAD/docs/run-s.md)) behaves the way I wanted, running everything no matter what, and reporting an error from any part.